examples/pennylane_run_doubly_stochastic.py

```diff
@@ -300,13 +315,73 @@ def moving_average(data, n=3):
 # correctly provides the ground state energy when evaluated for a larger
 # number of shots:
 
-print("Doubly stochastic gradient descent min energy = ", qnode_GD(params))
+print("Doubly stochastic gradient descent min energy = ", qnode_analytic(params))
+
+##############################################################################
+# While stochastic gradient descent requires more optimization steps to achieve
+# convergence, it is worth noting that it requires significantly fewer quantum
+# device evaluations, and thus may as a result take less time overall.
+
+##############################################################################
+# Adaptive stochasticity
+# ----------------------
+#
+# To improve on the convergence, we may even consider a crude "adaptive" modification
+# of the doubly stochastic gradient descent optimization performed above. In this
+# approach, we successively increase the number of terms we are sampling from as
+# the optimization proceeds, as well as increasing the number of shots.
+
+dev_stochastic.shots = 100
+cost = []
+params = init_params
+opt = qml.GradientDescentOptimizer(0.005)
+
+for i in range(250):
+
+    if i < 10:
+        n = 1
+        dev_stochastic.shots = 1
+    elif i < 25:
+        n = 2
+        dev_stochastic.shots = 10
+    elif i < 100:
+        n = 3
+        dev_stochastic.shots = 100
+    elif i < 150:
+        n = 4
+        dev_stochastic.shots = 500
+    elif i < 250:
+        n = 5
+        dev_stochastic.shots = 1000

```

Comment:
Now I see why you call it crude... What about shortening this part a bit by making it a queue-like object?

```python
adaptions = [(10, 1, 1), (25, 2, 10), (100, 3, 100), (150, 4, 500), (250, 5, 1000)]

threshold, n, shots = adaptions.pop(0)
for i in range(250):
    if i >= threshold:
        threshold, n, shots = adaptions.pop(0)

    ...
```
I obviously didn't think too much about the name `adaptions`.

Reply:
Maybe it would be better to use a strategy that adapts programmatically, like

```python
for i in range(250):
    n = int(i / 50) + 1
    shots = 1 + (n-1)**2 * 20
```